src/libmongoc/tests/test-mongoc-client-side-encryption.c

```diff
@@ -6103,6 +6103,136 @@ test_bypass_mongocryptd_shared_library (void *unused)
+
+      mongoc_client_encryption_datakey_opts_destroy (dkOpts);
+   }
+
+   bson_value_t minValue = {.value_type = BSON_TYPE_INT32, .value.v_int32 = 0};
+   bson_value_t maxValue = {.value_type = BSON_TYPE_INT32, .value.v_int32 = 1000};
+   bson_value_t toEncrypt = {.value_type = BSON_TYPE_INT32, .value.v_int32 = 123};
+
+   // Create `payload_defaults`.
+   bson_value_t payload_defaults;
+   {
+      mongoc_client_encryption_encrypt_opts_t *eOpts = mongoc_client_encryption_encrypt_opts_new ();
+      mongoc_client_encryption_encrypt_opts_set_keyid (eOpts, &keyID);
+      mongoc_client_encryption_encrypt_opts_set_algorithm (eOpts, MONGOC_ENCRYPT_ALGORITHM_RANGE);
+      mongoc_client_encryption_encrypt_opts_set_contention_factor (eOpts, 0);
+
+      // Apply range options. Omit `sparsity` and `trimFactor`.
+      {
+         mongoc_client_encryption_encrypt_range_opts_t *rOpts = mongoc_client_encryption_encrypt_range_opts_new ();
+         mongoc_client_encryption_encrypt_range_opts_set_min (rOpts, &minValue);
+         mongoc_client_encryption_encrypt_range_opts_set_max (rOpts, &maxValue);
+         mongoc_client_encryption_encrypt_opts_set_range_opts (eOpts, rOpts);
+         mongoc_client_encryption_encrypt_range_opts_destroy (rOpts);
+      }
+
+      bool ok = mongoc_client_encryption_encrypt (clientEncryption, &toEncrypt, eOpts, &payload_defaults, &error);
+      ASSERT_OR_PRINT (ok, error);
+
+      mongoc_client_encryption_encrypt_opts_destroy (eOpts);
+   }
+
+   // Case 1: Uses libmongocrypt defaults.
+   {
+      mongoc_client_encryption_encrypt_opts_t *eOpts = mongoc_client_encryption_encrypt_opts_new ();
+      mongoc_client_encryption_encrypt_opts_set_keyid (eOpts, &keyID);
+      mongoc_client_encryption_encrypt_opts_set_algorithm (eOpts, MONGOC_ENCRYPT_ALGORITHM_RANGE);
+      mongoc_client_encryption_encrypt_opts_set_contention_factor (eOpts, 0);
+
+      // Apply range options. Include `sparsity` and `trimFactor`.
+      {
+         mongoc_client_encryption_encrypt_range_opts_t *rOpts = mongoc_client_encryption_encrypt_range_opts_new ();
+         mongoc_client_encryption_encrypt_range_opts_set_min (rOpts, &minValue);
+         mongoc_client_encryption_encrypt_range_opts_set_max (rOpts, &maxValue);
+         mongoc_client_encryption_encrypt_range_opts_set_sparsity (rOpts, 2);
+         mongoc_client_encryption_encrypt_range_opts_set_trim_factor (rOpts, 6);
+         mongoc_client_encryption_encrypt_opts_set_range_opts (eOpts, rOpts);
+         mongoc_client_encryption_encrypt_range_opts_destroy (rOpts);
+      }
+
+      bson_value_t payload;
+      bool ok = mongoc_client_encryption_encrypt (clientEncryption, &toEncrypt, eOpts, &payload, &error);
+      ASSERT_OR_PRINT (ok, error);
+
+      // Assert both payloads have equal length. Intended to check they used the same `trimFactor` and `sparsity`.
+      ASSERT (payload.value_type == BSON_TYPE_BINARY);
+      ASSERT (payload_defaults.value_type == BSON_TYPE_BINARY);
```

Comment:
Can this assertion of `payload_defaults` be moved up to line 6165? It looks out-of-place here and below on line 6223.